source/common/upstream/upstream_impl.h

```diff
@@ -101,6 +102,57 @@ class UpstreamLocalAddressSelectorImpl : public UpstreamLocalAddressSelector {
   std::vector<UpstreamLocalAddress> upstream_local_addresses_;
 };
 
+/**
+ * Class for LBPolicies
+ * Uses a absl::variant to store pointers for the LBPolicy
+ */
+class LBPolicyConfig {
+public:
+  LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  template <typename T> OptRef<const T> getConfig() const {
+    if (const auto lbPtr = absl::get_if<std::unique_ptr<const T>>(&lb_policy_); lbPtr) {
+      if (*lbPtr != nullptr) {
```

Comment:
This condition can also be moved to the `if` statement above.
It will make the code more readable as there will be a single if-else.

Reply:
Done